components/region_cache_memory_engine/src/background.rs

```diff
@@ -37,11 +43,97 @@ fn parse_write(value: &[u8]) -> Result<WriteRef<'_>, String> {
     }
 }
 
+pub struct BackgroundWork {
+    worker: Worker,
+    scheduler: Scheduler<BackgroundTask>,
+
+    tick_stopper: Option<(JoinHandle<()>, Sender<bool>)>,
+}
+
+impl Drop for BackgroundWork {
+    fn drop(&mut self) {
+        let (h, tx) = self.tick_stopper.take().unwrap();
+        let _ = tx.send(true);
+        let _ = h.join();
+        self.worker.stop();
+    }
+}
+
+impl BackgroundWork {
+    pub fn new(core: Arc<ShardedLock<RangeCacheMemoryEngineCore>>, gc_interval: Duration) -> Self {
+        let worker = Worker::new("range-cache-background-worker");
+        let runner = BackgroundRunner::new(core.clone());
+        let scheduler = worker.start("range-cache-engine-background", runner);
+
+        let scheduler_clone = scheduler.clone();
+
+        let (handle, tx) = BackgroundWork::start_tick(scheduler_clone, gc_interval);
+
+        Self {
+            worker,
+            scheduler,
+            tick_stopper: Some((handle, tx)),
+        }
+    }
+
+    fn start_tick(
+        scheduler: Scheduler<BackgroundTask>,
+        gc_interval: Duration,
+    ) -> (JoinHandle<()>, Sender<bool>) {
+        let (tx, rx) = bounded(0);
+        let h = std::thread::spawn(move || {
+            loop {
+                select! {
+                    recv(tick(gc_interval)) -> _ => {
+                        if scheduler.is_busy() {
+                            info!(
```

Comment:
this may be too annoying, debug is ok.

Reply:
I think it's fine. Only when the scheduler is busy (ongoing gc is not finished) will this log be printed.